### PR TITLE
Crs 2192 t3 offence exclusions from early release

### PR DESCRIPTION
**Pull request by mg-moj**
Exclude sentences from early release where T3 related offences have a release date on or after the T3 commencement date.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/OffenceSDSReleaseArrangementLookupService.kt [hunk 1]
```diff
@@ -153,11 +153,24 @@ class OffenceSDSReleaseArrangementLookupService(
...
+      SDSEarlyReleaseExclusionSchedulePart.SEXUAL_T3,
+      -> SDSEarlyReleaseExclusionType.SEXUAL
+      SDSEarlyReleaseExclusionSchedulePart.DOMESTIC_ABUSE,
+      SDSEarlyReleaseExclusionSchedulePart.DOMESTIC_ABUSE_T3,
+      -> SDSEarlyReleaseExclusionType.DOMESTIC_ABUSE
+      SDSEarlyReleaseExclusionSchedulePart.NATIONAL_SECURITY,
+      SDSEarlyReleaseExclusionSchedulePart.NATIONAL_SECURITY_T3,
+      -> SDSEarlyReleaseExclusionType.NATIONAL_SECURITY
+      SDSEarlyReleaseExclusionSchedulePart.TERRORISM,
+      SDSEarlyReleaseExclusionSchedulePart.TERRORISM_T3,
```
Comment (ldlharper) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/OffenceSDSReleaseArrangementLookupService.kt:166]: Shouldn't these exclusion schedule parts be mapped to the equivalent T3 exclusion types?

Comment (mg-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/OffenceSDSReleaseArrangementLookupService.kt:166]: That's correct. I have these changes on a follow up ticket, but it makes more sense to add them to this branch. I'll make the commit shortly.

**Review by ldlharper — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SDSEarlyReleaseExclusionType.kt [hunk 2]
```diff
@@ -12,6 +12,13 @@ enum class SDSEarlyReleaseExclusionType {
   DOMESTIC_ABUSE,
   NATIONAL_SECURITY,
   TERRORISM,
+  SEXUAL_T3,
+  VIOLENT_T3,
+  DOMESTIC_ABUSE_T3,
+  NATIONAL_SECURITY_T3,
+  TERRORISM_T3,
+  MURDER,
```
Comment (joelstobart-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SDSEarlyReleaseExclusionType.kt:20]: There is no "old" murder list?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/external/manageoffencesapi/SDSEarlyReleaseExclusionForOffenceCode.kt [hunk 3]
```diff
@@ -7,5 +7,12 @@ enum class SDSEarlyReleaseExclusionSchedulePart {
   DOMESTIC_ABUSE,
   NATIONAL_SECURITY,
   TERRORISM,
+  SEXUAL_T3,
+  VIOLENT_T3,
+  DOMESTIC_ABUSE_T3,
+  NATIONAL_SECURITY_T3,
+  TERRORISM_T3,
+  MURDER,
```
Comment (joelstobart-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/external/manageoffencesapi/SDSEarlyReleaseExclusionForOffenceCode.kt:15]: again?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/timeline/handlers/TimelineCalculationHandler.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+abstract class TimelineCalculationHandler(
+  protected val trancheConfiguration: SDS40TrancheConfiguration,
+  protected val multiplierLookup: ReleasePointMultiplierLookup,
+  protected val timelineCalculator: TimelineCalculator,
+) {
+  abstract fun handle(timelineCalculationDate: LocalDate, timelineTrackingData: TimelineTrackingData): TimelineHandleResult
+
+  /**
+   * Return lambda function which receives the sentence track and returns the
```
Comment (joelstobart-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/timeline/handlers/TimelineCalculationHandler.kt:20]: This isn't a very useful comment for new developers, it's a bit ambiguous.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/timeline/handlers/TimelineCalculationHandler.kt [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+   * sentence multiplier for use when calculating the earliest release date.
+   *
+   * 1. If timelineCalculationDate if on or after tranche three commencement date and
```
Comment (joelstobart-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/timeline/handlers/TimelineCalculationHandler.kt:23]: multiple "if"s on this line makes it confusing

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/timeline/handlers/TimelineCalculationHandler.kt [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+   *    the track is SDS_STANDARD_RELEASE_T3_EXCLUSION, then return default SDS_STANDARD multiplier,
+   *    as related offence is now excluded from early release
+   * 2. If timelineCalculationDate date is on or after trancheOneCommencementDate, return multiplier which
+   *    may be eligible for early release (early release scheme commenced as of tranche one)
+   * 3. Default to historic multipliers, which are pre early release
```
Comment (joelstobart-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/timeline/handlers/TimelineCalculationHandler.kt:28]: It's not early release - it's prior to the SDS40 release arrangement rules

**Pull request merged**